File: test/unit/algebra_test.rb
#!/usr/bin/env ruby
# encoding: utf-8

$use_tree=true
$use_tree=false

require_relative '../parser_test_helper'

class AlgebraTest < ParserBaseTest

  include ParserTestHelper

  def test_algebra1
    assert_result_is "two minus 1½",0.5
    assert_result_is "3 minus one",2
    init "4½"
    assert_equals @parser.fraction,4.5
    init "4½+3½"
    @parser.do_interpret!
    assert_equals @parser.algebra,8
    assert_result_is "4½+3½","8"
  end


  def test_algebra_NOW
    skip "test_algebra_NOW, DONT SKIP!"
    assert_result_is "1+(3/4)","1¾"
    assert_result_is "1+3/4.0","1¾"
    # 1.0 noo
    assert_result_is "1.0+3/4.0","1¾"
    # 4.0
  end

  def test_algebra
    ok=parse "2*(3+10)"
    puts "Parsed input as #{ok}!"
    assert_equals ok,26
    skip
    assert @current_node!=nil
    #assert @current_node==@root
    full_value=@current_node.full_value
    val=eval(full_value)
    assert_equals val,26
    val=@current_node.eval_node(@variableValues)
    assert_equals val,26
  end

end

File: angelos/test/algebra_test.py
import _global
_global.use_tree = True
_global.use_tree = False
from parser_test_helper import *


class AlgebraTest(ParserBaseTest):
    
    def test_algebra1(self):
        assert_result_is(u'two minus 1\xbd', None)
        assert_result_is('3 minus one', 2)
        init(u'4\xbd')
        assert_equals(self.parser.fraction(), None)
        init(u'4\xbd+3\xbd')
        self.parser.do_interpret()
        assert_equals(self.parser.algebra(), 8)
        assert_result_is(u'4\xbd+3\xbd', '8')

    def test_algebra_NOW(self):
        skip('test_algebra_NOW, DONT SKIP!')
        assert_result_is('1.0+(3/4.0)', 7/4)
        assert_result_is('1.0+3/4.0', 7/4)
        assert_result_is('1+3/4.0', 7/4)
        assert_result_is('1+(3/4)', 7/4)

    def test_algebra(self):
        ok = parse('2*(3+10)')
        print((('Parsed input as ' + str(ok)) + '!'))
        assert_equals(ok, 26)
        # skip()
        # assert(self.current_node!=(None))
        # full_value = self.current_node.full_value()
        # val = eval(full_value)
        # assert_equals(val, 26)
        # val = self.current_node.eval_node(self.variableValues)
        # assert_equals(val, 26)